simulator-ui/tests/entities.spec.ts

```diff
@@ -0,0 +1,251 @@
+    locators: [
+      'th :text("Name")',
+      'th :text("Start Date")',
+      'th :text("End Date")',
+      'th :text("Status")',
+      'th :text("Error Message")',
+    ],
+    testIdsAndExpectedValues: [
+      {id: 'scenarioExecutionEntityScenarioExecutionLink', value: '1'},
+      {id: 'scenarioExecutionEntityScenarioName', value: 'Default'},
+      {id: 'scenarioExecutionEntityStartDate', value: exampleDate},
+      {id: 'scenarioExecutionEntityEndDate', value: exampleDate},
+      {id: 'scenarioExecutionEntityStatus', value: 'FAILURE'},
+      {id: 'scenarioExecutionEntityTestResult', value: 'New Error'},
+    ],
+    testIdToBeVisible: []
+  },
+  {
+    apiUrl: '**/api/scenario-actions*',
+    entityUrl: 'http://localhost:9000/scenario-action',
+    contentJson: scenarioActionJson,
+    locators: ['th :text("Name")', 'th :text("Start Date")', 'th :text("End Date")', 'th :text("Scenario Execution")'],
+    testIdsAndExpectedValues: [
+      {id: 'scenarioActionEntitiesId', value: '1'},
+      {id: 'scenarioActionEntitiesName', value: 'http:receive-request'},
+      {id: 'scenarioActionEntitiesStartDate', value: exampleDate},
+      {id: 'scenarioActionEntitiesEndDate', value: exampleDate},
+    ],
+    testIdToBeVisible: []
+  },
+  {
+    apiUrl: '**/api/test-results*',
+    entityUrl: 'http://localhost:9000/test-result',
+    contentJson: testResultJson,
+    locators: [
+      'th :text("Status")',
+      'th :text("Test Name")',
+      'th :text("Class Name")',
+      'th :text("Error Message")',
+      'th :text("Stack Trace")'],
+    testIdsAndExpectedValues: [
+      {id: 'testResultEntitiesId', value: '1'},
+      {id: 'testResultEntitiesStatus', value: 'FAILURE'},
+      {id: 'testResultEntitiesTestName', value: 'Scenario(Default)'},
+      {id: 'testResultEntitiesClassName', value: 'DefaultTestCase'},
+      {id: 'testResultEntitiesErrorMessage', value: 'New Error'},
+      {id: 'testResultEntitiesStackTrace', value: 'New Stacktrace'},
+      {id: 'testResultEntitiesFailureType', value: ''},
+      {id: 'testResultEntitiesCreatedDate', value: exampleDate},
+      {id: 'testResultEntitiesLastModifiedDate', value: exampleDate},
+    ],
+    testIdToBeVisible: []
+  },
+  {
+    apiUrl: '**/api/test-parameters*',
+    entityUrl: 'http://localhost:9000/test-parameter',
+    contentJson: testParameterJson,
+    locators: [
+      'th :text("Key")',
+      'th :text("Value")',
+      'th :text("Test Result")',
+      'th :text("Created Date")',
+      'th :text("Last Modified Date")'
+    ],
+    testIdsAndExpectedValues: [
+      {id: 'testParameterEntityKey', value: 'test key'},
+      {id: 'testParameterEntityValue', value: 'test value'},
+      {id: 'testParameterEntityTestResultLink', value: '0'},
+      {id: 'testParameterEntityCreatedDate', value: exampleDate},
+      {id: 'testParameterEntityLastModifiedDate', value: exampleDate},
+    ],
+    testIdToBeVisible: ['', '']
+  }
+]
+
```

Comment:
the tests are very repetitively; is it better to put them into a for loop to avoid redundant code? Or should the tests be put into separate files 'one file per page'?